community/kernel-api/src/main/java/org/neo4j/internal/kernel/api/ExplicitIndexWrite.java

```diff
@@ -22,24 +22,131 @@
 import java.util.Map;
 
 import org.neo4j.internal.kernel.api.exceptions.KernelException;
+import org.neo4j.internal.kernel.api.exceptions.explicitindex.ExplicitIndexNotFoundKernelException;
 
 /**
  * Operations for creating and modifying explicit indexes.
  */
 public interface ExplicitIndexWrite
 {
+    /**
+     * Adds node to explicit index.
+     *
+     * @param indexName The name of the index
+     * @param node The id of the node to add
+     * @param key The key to associate with the node
+     * @param value The value to associate with the node an key
+     * @throws ExplicitIndexNotFoundKernelException If there is no explicit index with the given name
+     */
+    void nodeAddToExplicitIndex( String indexName, long node, String key, Object value )
+            throws KernelException;
+
+    /**
+     * Removes a node from an explicit index
+     *
+     * @param indexName The name of the index
+     * @param node The id of the node to remove
+     * @param key The key to associate with the node
+     * @param value The value to associate with the node and key
```

Comment:
I'm sorry. I should have been more specific from the start. Third time's the charm.

@param key The key ~to associate~ associated with the node
@param value The value ~to associate~ associated with the node and key